packages/node-runtime-worker-thread/src/rpc.ts

```diff
@@ -7,12 +7,44 @@ import {
   ServerMessageData,
   ClientMessageData
 } from 'postmsg-rpc';
+import { deserializeError, serializeError } from './serializer';
+
+function serialize(data: unknown): string {
+  return `data:;base64,${v8.serialize(data).toString('base64')}`;
+}
+
+function deserialize<T = unknown>(str: string): T | string {
+  if (/^data:;base64,.+/.test(str)) {
+    return v8.deserialize(
+      Buffer.from(str.replace('data:;base64,', ''), 'base64')
+    );
+  }
+  return str;
+}
 
 type RPCMessageBus = { on: Function; off: Function } & (
   | { postMessage: Function; send?: never }
   | { postMessage?: never; send?: Function }
 );
 
+const ERROR = '$$ERROR';
+
+const MESSAGE = '$$MESSAGE';
```

Comment:
Super-hyper minor .. what about we qualify the name here `TYPE_MESSAGE` and `TYPE_ERROR` or similar?

Reply:
Sure, will do